docs/controllers/optimization.md

```diff
@@ -0,0 +1,224 @@
+# Optimization
+
+This document aims to help optimize various factors of resource consumption by controllers, and it will effectively be a guide on how to reduce the usage of your [watcher] streams to simplify things for downstream consumers.
+
+## Watcher Streams in Controllers
+
+By **default**, [watcher] streams are **implicitly configured** within the [Controller], but using the controller streams interface setup introduced in [kube 0.81](https://github.com/kube-rs/kube/releases/tag/0.81.0) you can **explicitly setup** all the [watcher] stream for more precise targeting:
+
+```rust
+let cfg = watcher::Config { RESTRICTIONS };
+let stream = reflector(writer, SOME_WATCHER(cfg))
+Controller::for_stream(stream, reader);
+```
+
+As this document will show, there are various ways to put restrictions on your watcher (and there are also two main types of watchers), but the central premise herein is that **controllers benefit from customizing the input stream** and we will show how to do these configurations.
+
+!!! warning "The controller streams interface is unstable"
+
+    Currently plugging streams into [Controller] requires the `kube/unstable-runtime` feature. This feature requirement will be removed in a future release.
+
+The controller streams interface is comprised of [Controller::for_stream], and [Controller::watches_stream] and [Controller::owns_stream], which are stream input analogues for the commonly advertised `Controller::new`, `Controller::watches`, and `Controller::owns` (respectively) interfaces.
+
+## Watcher Optimization
+
+One of the biggest contributor to activity in a [Controller] is the constant, long-polling watch of the [[object]] and every related object by the use of multiple [watcher] streams.
+
+By themselves, the optimizations listed herein for watchers are generally for reducing IO or networked traffic. However, when used in combination with [reflector] caches they also become memory optimizations.
+
+
+### Reducing Number of Watched Objects
+The default `watcher::Config` will watch every object in the [Api] scope you configured:
+
+- `Api::namespaced` or `Api::default_namespaced` -> all objects in that namespace
+- `Api::all` -> all cluster scoped objects (or all objects in all namespaces)
+
+This can be limited to just a subset of namespaces, or a subset of objects using field selectors to limit to a selection of known names:
```

Comment:
```suggestion
This can be limited to just a subset of namespaces, or other properties on the objects. For example, field selectors can be used to limit to a selection of known names:
```